### PR TITLE
fix: accept new content-type downloading the platform

### LINKED ISSUE #63: Error downloading a valid Rider version
Hello,

I got the following error adding the `riderRD:2022.2` ide version.

```
  Error: =======================================================================================
  Error: It appears the download of https://www.jetbrains.com/intellij-repository/releases/com/jetbrains/intellij/rider/riderRD/2022.2/riderRD-2022.2.zip did not contain the following:
  Error:     - status: 200
  Error:     - content-type: application/octet-stream
  Error: Actual response: HTTP/200 - content-type: application/x-zip-compressed
  Error: This can happen if riderRD:2022.2 is not a valid IDE / version. If you believe it is a
  Error: valid ide/version, please open an issue on GitHub:
  Error:      https://github.com/ChrisCarini/intellij-platform-plugin-verifier-action/issues/new
  Error: As a precaution, we are failing this execution.
  Error: =======================================================================================
  Error: Exiting due to a known, handled exception - invalid download headers when downloading an IDE.
```

Might be because of the content-type being `application/x-zip-compressed` (other working ide versions have a `application/octet-stream` content-type).

Regards,
Yann

### PR DESCRIPTION
**Pull request by le-yams**
fixes #63

### CODE REVIEW

**Comment by le-yams** [comment]
@ChrisCarini I made it possible to add new content-type if another case present itself :)

**Comment by ChrisCarini** [comment]
> @ChrisCarini I made it possible to add new content-type if another case present itself :)

Nice! This looks great! 🎉 

Before merging this in, are you able to share a link to an action run with this change? You should be able to temporarily change a workflow that uses this to point to your fork+branch (e.g. `le-yams/intellij-platform-plugin-verifier-action@fix/download_platform_accepted_content_type`). If you're not able to share a link (perhaps private repo/project), could you just include a screenshot of the respective output for Rider IDE (since that is where you found this bug)? Just want to confirm (mostly for posterity) this works as expected.

Thank you! Excited to get this merged in, and thank you for (a) using this GitHub Action, and, more importantly, (b) contributing back via a bug report + fix! 🎉 👏

**Comment by le-yams** [comment]
> Before merging this in, are you able to share a link to an action run w

Yes, I thought about it just after creating the PR. I'll come back as soon as I have a green action :)

**Comment by le-yams** [comment]
I got an error with the if statement:
```bash
if [[ ${http_status} --eq 200 ]] && is_valid_platform_response_content_type "${content_type}"; then
```
gives me
> conditional binary operator expected

My bash is a bit rusty ans it works fine on my computer, do you have any idea where I get it wrong?

**Comment by le-yams** [comment]
Fyi: i'm testing with this pr (public project): https://github.com/le-yams/openfga4intellij/pull/4/checks

File: entrypoint.sh [hunk 1]
```diff
@@ -320,23 +334,23 @@ echo "$INPUT_IDE_VERSIONS" | while read -r IDE_VERSION; do
...
+  content_type==$(echo "${CURL_RESP}" | jq -r '.content_type // empty')
 
   gh_debug "Checking response code and content type for the download of [$DOWNLOAD_URL] to ensure download successful..."
-  # Turn off 'exit on error'; if we error out when testing the response code,
-  # we want to first print a friendly message to the user and then exit.
-  set +o errexit
-  echo "$CURL_RESP" | grep "HTTP/200 - content-type: application/octet-stream"
-  if [[ $? -eq 0 ]]; then
+
+  if [[ ${http_status} --eq 200 ]] && is_valid_platform_response_content_type "${content_type}"; then
```
Comment (ChrisCarini) [inline entrypoint.sh:344]:
```suggestion
  if [ "$http_status" -eq 200 ] && is_valid_platform_response_content_type "${content_type}"; then
```

this seems to work for me:

Screenshot 2023-11-16 at 07 44 05


fwiw, works under both versions of bash I tested against:
```
$ > /opt/homebrew/bin/bash test.sh 
Bash version: 5.2.15(1)-release
GOOD

$ > /bin/bash test.sh 
Bash version: 3.2.57(1)-release
GOOD
```

@le-yams - mind giving this a try just to confirm it works on the GitHub Actions runners?

File: entrypoint.sh [hunk 2]
```diff
@@ -320,23 +334,23 @@ echo "$INPUT_IDE_VERSIONS" | while read -r IDE_VERSION; do
   ZIP_FILE_PATH="$HOME/$IDE-$VERSION.zip"
 
   echo "Downloading $IDE $IDE_VERSION from [$DOWNLOAD_URL] into [$ZIP_FILE_PATH]..."
-  CURL_RESP=$(curl -L --silent --show-error -w 'HTTP/%{http_code} - content-type: %{content_type}' --output "$ZIP_FILE_PATH" "$DOWNLOAD_URL")
+  
+  CURL_RESP=$(curl -L --silent --show-error -w '%{json}' --output "$ZIP_FILE_PATH" "$DOWNLOAD_URL")
+  http_code==$(echo "${CURL_RESP}" | jq -r '.response_code // empty')
+  content_type==$(echo "${CURL_RESP}" | jq -r '.content_type // empty')
```
Comment (ChrisCarini) [inline entrypoint.sh:340]:
```suggestion
  http_code=$(echo "${CURL_RESP}" | jq -r '.response_code // empty')
  content_type=$(echo "${CURL_RESP}" | jq -r '.content_type // empty')
```

you're getting an extra `=` prepended to these two vars.

```
$ > cat test.sh 
#!/bin/bash


CURL_RESP=$(curl -L --silent --show-error -w '%{json}' --output "ideaIC-2022.2.zip" https://www.jetbrains.com/intellij-repository/releases/com/jetbrains/intellij/idea/ideaIC/2022.2/ideaIC-2022.2.zip)

echo $CURL_RESP

http_code==$(echo "${CURL_RESP}" | jq -r '.response_code // empty')
content_type==$(echo "${CURL_RESP}" | jq -r '.content_type // empty')

echo "${http_code} - ${content_type}"
```

output:
```
..."curl_version":"libcurl/8.1.2 (SecureTransport) LibreSSL/3.3.6 zlib/1.2.11 nghttp2/1.51.0"}
=200 - =application/octet-stream
```

**Comment by le-yams** [comment]
I fixed the http code comparison but now the script is failing without very much details :( 
>  Error: =======================================================
  Error: ===-------------------------------------------------===
  Error: === An unexpected error occurred. Status code [1].
  Error: ===
  Error: === Please consider opening a bug for the developer:
  Error: ===    - Bug URL: https://github.com/ChrisCarini/intellij-platform-plugin-verifier-action/issues/new
  Error: ===    - Status Code: 1
  Error: ===-------------------------------------------------===
  Error: =======================================================
  
  cf the action run [here](https://github.com/le-yams/openfga4intellij/actions/runs/6893520588/job/18753479199?pr=4)

**Comment by ChrisCarini** [comment]
> I fixed the http code comparison but now the script is failing without very much details :(
> 
> > Error: =======================================================
> > Error: ===-------------------------------------------------===
> > Error: === An unexpected error occurred. Status code [1].
> > Error: ===
> > Error: === Please consider opening a bug for the developer:
> > Error: ===    - Bug URL: https://github.com/ChrisCarini/intellij-platform-plugin-verifier-action/issues/new
> > Error: ===    - Status Code: 1
> > Error: ===-------------------------------------------------===
> > Error: =======================================================
> 
> cf the action run [here](https://github.com/le-yams/openfga4intellij/actions/runs/6893520588/job/18753479199?pr=4)

try my suggestion here -> https://github.com/ChrisCarini/intellij-platform-plugin-verifier-action/pull/66/files#r1395994027

I actually think that the check you currently have is incorrectly failing and giving you ^ error message now. It's comparing `"200" = "=200"`, and thus failing

**Comment by le-yams** [comment]
> > I fixed the http code comparison but now the script is failing without very much details :(
> > > Error: =======================================================
> > > Error: ===-------------------------------------------------===
> > > Error: === An unexpected error occurred. Status code [1].
> > > Error: ===
> > > Error: === Please consider opening a bug for the developer:
> > > Error: ===    - Bug URL: https://github.com/ChrisCarini/intellij-platform-plugin-verifier-action/issues/new
> > > Error: ===    - Status Code: 1
> > > Error: ===-------------------------------------------------===
> > > Error: =======================================================
> > 
> > 
> > cf the action run [here](https://github.com/le-yams/openfga4intellij/actions/runs/6893520588/job/18753479199?pr=4)
> 
> try my suggestion here -> https://github.com/ChrisCarini/intellij-platform-plugin-verifier-action/pull/66/files#r1395994027
> 
> I actually think that the check you currently have is incorrectly failing and giving you ^ error message now. It's comparing `"200" = "=200"` _(note the extra `=` in the second string)_, and thus failing

sorry I missed your suggestion (and this typo ^^)

**Comment by ChrisCarini** [comment]
> @ChrisCarini [it worked!](https://github.com/le-yams/openfga4intellij/actions/runs/6893858991/job/18754374718?pr=4) \o/ Thanks for your help :)

Nice! 🎉 👏

**Review by ChrisCarini — APPROVED** [review]
LGTM!

**Comment by ChrisCarini** [comment]
@all-contributors please add @le-yams for bug and code

**Pull request merged**